test/solvers/functional/default_functionals_test.py

```diff
@@ -0,0 +1,319 @@
+    # Explicit calculation: ||x||^2 / 4
+    expected_result = x.norm()**2 / 4.0
+    assert almost_equal(cc_func(x), expected_result, places=PLACES)
+
+    # The gradient of the convex conjugate (not implemeted)
+    # Explicit calculation: x / 2
+    expected_result = x / 2.0
+    assert all_almost_equal(cc_func.gradient(x), expected_result,
+                            places=PLACES)
+
+    # The proximal of the convex conjugate
+    # Explicit computation: x / (1 + sigma/2)
+    expected_result = x / (1 + sigma / 2.0)
+    assert all_almost_equal(cc_func.proximal(sigma)(x), expected_result,
+                            places=PLACES)
+
+    # The biconjugate, which is the functional itself since it is proper,
+    # convex and lower-semicontinuous
+    cc_cc_func = cc_func.convex_conj
+
+    # Modulo scaling back and forth, check that it is in fact squared L2-norm
+    assert isinstance(cc_cc_func.orig_cconj_f.operator,
+                      odl.solvers.L2NormSquared)
+
+    # Check that they evaluate the same
+    assert almost_equal(cc_cc_func(x), func(x), places=PLACES)
+
+    # Check that they evaluate the gradient the same
+    assert all_almost_equal(cc_cc_func.gradient(x), func.gradient(x),
+                            places=PLACES)
+
+
+def test_constant_functional():
+    """Test the constant functional."""
+
+    n = 10
+    space = odl.rn(n)
+    constant = np.random.randn()
+    func = odl.solvers.ConstantFunctional(space, constant=constant)
+    x = noise_element(space)
+
+    # Checking that constant is stored correctly
+    assert func.constant == constant
+
+    # Evaluation of the functional
+    expected_result = constant
+    assert almost_equal(func(x), expected_result, places=PLACES)
+
+    # The gradient
+    # Given by the zero-operator
+    assert isinstance(func.gradient, odl.ZeroOperator)
+
+    # The proximal operator
+    sigma = np.random.rand()
+    # This is the identity operator
+    assert isinstance(func.proximal(sigma), odl.IdentityOperator)
+
+    # The convex conjugate functional
+    cc_func = func.convex_conj
+
+    # Evaluation of convex conjugate
+    # Explicit calculation: -constant if x=0, infty else
+    if not x.norm() == 0:
+        assert cc_func(x) == np.inf
+    else:
+        assert cc_func(x) == -constant
+
+    assert cc_func(space.zero()) == -constant
+
+    # The gradient of the convex conjugate (not implemeted)
+    with pytest.raises(NotImplementedError):
+        cc_func.gradient
+
+    # The proximal of the convex conjugate
+    sigma = np.random.rand()
+    # This is the zero operator
+    assert isinstance(cc_func.proximal(sigma), odl.ZeroOperator)
+
+    # The biconjugate, which is the functional itself since it is proper,
+    # convex and lower-semicontinuous
+    expected_functional = odl.solvers.ConstantFunctional(space, constant)
+    cc_cc_func = cc_func.convex_conj
+    assert isinstance(cc_cc_func, type(expected_functional))
+
+
+def test_zero_functional():
```

Comment:
i.m.o. the zero functional has a trivial implementation and does not really need a test. At most, verify that 

```python
isinstance(ZeroFunctional(space), ConstantFunctional)
assert ZeroFunctional(space).constant = 0
```

everything else follows from the test above.